providers/NFTNameGeneratorProvider.tsx

```diff
@@ -0,0 +1,188 @@
+
+    // eslint-disable-next-line no-plusplus
+    while (n--) {
+      u8arr[n] = bstr.charCodeAt(n)
+    }
+    return new File([u8arr], filename, { type: mime })
+  }
+
+  const checkTx = useCallback(async () => {
+    const response = await axios.get("/api/getTxLogs", {
+      params: {
+        txHash,
+      },
+    })
+    if (response) {
+      setTokenId(parseInt(response.data.logs[0].topics[3], 16).toString())
+      setHaveTokenId(true)
+    }
+  }, [txHash])
+
+  const postTweet = useCallback(
+    async (generatedName) => {
+      const response = await axios.post("/api/tweet", {
+        twitterHandle,
+        generatedName,
+      })
+      return response
+    },
+    [twitterHandle],
+  )
+
+  const getGeneratedName = useCallback(async () => {
+    const response = await axios.get("/api/randomName")
+    return response.data
+  }, [])
+
+  const generateImageURI = useCallback(
+    async (generatedName) =>
+      textToImage.generate(generatedName, {
+        debug: true,
+        fontSize: 58,
+        fontFamily: "Aileron",
+        lineHeight: 58,
+        margin: 5,
+        customHeight: 500,
+        maxWidth: 500,
+        bgColor: "black",
+        textColor: "white",
+        textAlign: "center",
+        verticalAlign: "center",
+      }),
+    [],
+  )
+
+  const mintGeneratedName = useCallback(
+    async (ipfsUrl: string, generatedName: string, creator: string) => {
+      const reciept = await axios.get("/api/mint", {
+        params: {
+          imageUri: ipfsUrl,
+          name: generatedName,
+          description: `${generatedName} generated by ${creator}`,
+        },
+      })
+      return reciept
+    },
+    [],
+  )
+  const handleGenerateClick = useCallback(async () => {
+    setLoading(true)
+    const generated = await getGeneratedName()
+    setName(generated)
+
+    const dataUri = await generateImageURI(generated)
+    setImageURI(dataUri)
+
+    const file = dataURLtoFile(dataUri, "a.png")
+    const ipfsUrl = await storeBlob(file)
+
+    const generatedBy = transformTwitterHandle(twitterHandle)
+    setDescription(`${generated} generated by ${generatedBy}`)
+    const receipt = await mintGeneratedName(ipfsUrl, generated, generatedBy)
+    if (receipt?.data?.txReceipt) {
+      setTxHash(receipt.data.txReceipt)
+    }
+    if (receipt.status === 200) {
+      setStartConfetti(true)
+      setTimeout(() => {
+        setStartConfetti(false)
+        setShowResults(true)
+      }, 5000)
+    }
+    const tweetResponse = await postTweet(generated)
+
+    if (tweetResponse?.data?.data?.id) {
+      setTwitterId(tweetResponse.data.data.id)
+    }
+    setLoading(false)
+  }, [twitterHandle, postTweet, getGeneratedName, generateImageURI, mintGeneratedName])
+
+  useEffect(() => {
+    let timeout = null
+    if (txHash.length > 0 && !haveTokenId) {
+      timeout = setTimeout(() => {
+        checkTx()
+      }, 5000)
+    }
+    return () => timeout && clearTimeout(timeout)
```

Comment:
just polling the blockscan every 5 seconds to see if transaction is completed. If so then I get token id from the logs

Reply:
interesting.......

what do you think of using the `wagmi` hook instead of polling?
- `useWaitForTransaction`

https://wagmi.sh/react/hooks/useWaitForTransaction

Reply:
<img width="875" alt="Screenshot 2022-12-24 at 10 10 15 AM" src="https://user-images.githubusercontent.com/23249402/209443662-8a379442-1630-4ecc-8db4-6ec968d23c3f.png">

Reply:
didnt know about this. I'll look into this